code/game/objects/structures/mirror.dm

```diff
@@ -156,6 +156,16 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror/broken, 28)
 		return TRUE
 
 	var/datum/species/newrace = selectable_races[racechoice]
+
+	newrace = new newrace()
```

Comment:
```suggestion
	newrace = new
```

however i get if that's a bit quirky so you can also do

```suggestion
	newrace = new()
```

but the first is preferred

Reply:
i tried that and it broke because it just created a /datum/species?

Reply:
OH right because it assumes that the typepath is implicit in the scenario, the way you have it written states it's explicit. the real way to do it then would be

```suggestion
	var/datum/species/newrace = new selectable_races[racechoice]
```